src/compile/layer.ts

```diff
@@ -0,0 +1,228 @@
+    });
+  }
+
+  private _initConfig(specConfig: Config, parent: Model) {
+    return mergeDeep(duplicate(defaultConfig), specConfig, parent ? parent.config() : {});
+  }
+
+  public has(channel: Channel): boolean {
+    // layer does not have any channels
+    return false;
+  }
+
+  public children() {
+    return this._children;
+  }
+
+  private hasSummary() {
+    // TODO: don't just use the first child
+    const summary = this._children[0].component.data.summary;
+    for (let i = 0; i < summary.length; i++) {
+      if (keys(summary[i].measures).length > 0) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public isOrdinalScale(channel: Channel) {
+    // since we assume shared scales we can just ask the first child
+    return this._children[0].isOrdinalScale(channel);
+  }
+
+  public dataTable(): string {
+    // TODO: think about what this should return, used by layout
+    return (this.hasSummary() ? SUMMARY : SOURCE) + '';
+  }
+
+  public fieldDef(channel: Channel): FieldDef {
+    return null; // layer does not have field defs
+  }
+
+  public stack() {
+    return null; // this is only a property for UnitModel
+  }
+
+  public parseData() {
+    this._children.forEach((child) => {
+      child.parseData();
+    });
+    this.component.data = parseLayerData(this);
+    console.log(this.component.data);
+  }
+
+  public parseSelectionData() {
+    // TODO: @arvind can write this
+    // We might need to split this into compileSelectionData and compileSelectionSignals?
+  }
+
+  public parseLayoutData() {
+    // TODO: correctly union ordinal scales rather than just using the layout of the first child
+    this._children.forEach((child, i) => {
+      child.parseLayoutData();
+    });
+    this.component.layout = parseLayerLayout(this);
+  }
+
+  public parseScale() {
+    const model = this;
+
+    let scaleComponent = this.component.scale = {} as Dict<ScaleComponent[]>;
+
+    this._children.forEach(function(child) {
+      child.parseScale();
+
+      // TODO: correctly implement independent scale
+      if (true) { // if shared/union scale
+        keys(child.component.scale).forEach(function(channel) {
+          let newScales = child.component.scale[channel] as ScaleComponent[];
+          if (!newScales) {
+            return;
+          }
+
+          const existingScales = scaleComponent[channel];
+          if (existingScales && newScales[0].type === ScaleType.ORDINAL) {
+            // Ordinal scales are unioned by combining the domain of the first scale. Other scales are appended.
+            const existingDomain = existingScales[0].domain.fields ? existingScales[0].domain : { fields: [existingScales[0].domain] };
+            newScales[0].domain = {
+              // assumes that the new scale domain is not a list
+              fields: existingDomain.fields.concat([newScales[0].domain])
+            };
+            if (existingScales.length > 1) {
+              delete existingScales[0];
```

Comment:
better write comment what's happening here.  why is this guy removed.  

I assume this case is currently only for color? 
Maybe worth changing the order in `scale.ts` so that `scales[0]` is always the main color scale and `[1]` is `colorLegend` ...